src/15-budget-intent/index.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * This experiment extracts a structured budget intent from user-supplied free text.
+ *
+ * Running this will generate multiple results across a number of examples and models.
+ *
+ * The results can be view in budgetIntents.ndjson
+ *
+ * They can also be converted to csv (via csv-kit) with:
+ *
+ * $ cat budgetIntents.ndjson | in2csv -f ndjson > budgetIntents.csv
+ *
+ * These^ csv results and some stats can be viewed at:
+ * https://docs.google.com/spreadsheets/d/1S1-7IJOPc26cgSlWaxP1bz9HEuNJXgQJTDdAyx7XRB0/edit?usp=sharing
+ *
+ */
+import _ from "lodash"
+import { anthropic } from "@ai-sdk/anthropic"
+import { LanguageModel, generateObject } from "ai"
+import { openai } from "@ai-sdk/openai"
+import { z } from "zod"
+import chalk from "chalk"
+import dedent from "dedent"
+import dotenv from "dotenv"
+import fs from "fs"
+import pAll from "p-all"
+import pSeries from "p-series"
+
+dotenv.config()
+
+const MODELS: Record<string, LanguageModel> = {
+  turbo: openai("gpt-3.5-turbo"),
+  fourO: openai("gpt-4o"),
+  haiku: anthropic("claude-3-haiku-20240307"),
+  // sonnet: anthropic("claude-3-sonnet-20240229"),
+  sonnet35: anthropic("claude-3-5-sonnet-20240620"),
```

Comment:
Hot off the presses: https://www.anthropic.com/news/claude-3-5-sonnet

Who can keep up? 😵‍💫